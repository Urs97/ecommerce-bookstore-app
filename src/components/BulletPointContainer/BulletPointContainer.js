import React from 'react';
import './BulletPointContainer.scss';
import BulletPoint from '../BulletPoint/BulletPoint';

const BulletPointContainer = ({ mode, icons, titles, texts, color }) => {

    if(mode === "regular") {
        return (
            <section className="bulletpoints">
                <BulletPoint mode ={mode} icon={icons[0]} text={texts[0]} color={color}/>
                <BulletPoint mode ={mode} icon={icons[1]} text={texts[1]} color={color}/>
                <BulletPoint mode ={mode} icon={icons[2]} text={texts[2]} color={color}/>
            </section>
        )
    } 
    
    else if (mode === "expanded") {
        return (
            <section className="bulletpoints">
                <BulletPoint mode={mode} icon={icons[0]} title={titles[0]} text={texts[0]} color={color}/>
                <BulletPoint mode={mode} icon={icons[1]} title={titles[1]} text={texts[1]} color={color}/>
                <BulletPoint mode={mode} icon={icons[2]} title={titles[2]} text={texts[2]} color={color}/>
            </section>
        )
    }
    
};

export default BulletPointContainer;