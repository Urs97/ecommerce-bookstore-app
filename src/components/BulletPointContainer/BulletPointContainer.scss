@import '../../styles/shared.scss';

.bulletpoints {
    display: flex;
    justify-content: space-between;
    padding: $vertical-content-padding $horizontal-content-padding;
}

@media (max-width: 1100px) {
    .bulletpoints {
        padding: $vertical-content-padding $horizontal-content-padding-1100px;
    }
}

@media (max-width: 950px) {
    .bulletpoints {
        padding: $vertical-content-padding $horizontal-content-padding-950px;
    }
}

@media (max-width: 865px) {
    .bulletpoints {
        padding: $vertical-content-padding $horizontal-content-padding-865px;
    }
}

// Mobile design

@media (max-width: 770px) {
    .bulletpoints {
        padding: $vertical-content-padding 1rem;
    }
}

@media (max-width: 750px) {
    .bulletpoints.expanded {
        display: flex;
        flex-direction: column;
        padding: $vertical-content-padding 10rem;

        .bulletpoint:not(:last-child) {
            margin-bottom: 4.5em;
        }
    }
}

@media (max-width: 580px) {
    .bulletpoints.expanded {
        padding: $vertical-content-padding 5rem;
    }
}

@media (max-width: 580px) {
    .bulletpoints {
        display: flex;
        flex-direction: column;
        padding: $vertical-content-padding 8rem;

        .bulletpoint:not(:last-child) {
            margin-bottom: 4.5em;
        }
    }
}

@media (max-width: 450px) {
    .bulletpoints {
        padding: $vertical-content-padding 5rem;
    }
}