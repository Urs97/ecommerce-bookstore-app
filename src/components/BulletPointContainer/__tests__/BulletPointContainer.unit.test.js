import React from 'react';
import {shallow} from 'enzyme';
import BulletPointContainer from '../BulletPointContainer';

const texts = ["This is text number 1", "This is text number 2", "This is text number 3"];
const titles = ["Title 1, Title 2, Title 3"];
const icons = ["mockIcon1.svg", "mockIcon2.svg", "mockIcon3.svg"]

describe('BulletPointContainer', () => {
    test('renders BulletPointContainer in unrecognizable mode', () => {
        const wrapper = shallow(
            <BulletPointContainer 
                mode="wrong-mode"
                icons={icons}
                titles={titles}
                texts={texts}
                color="yellow"/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders BulletPointContainer in normal mode', () => {
        const wrapper = shallow(
            <BulletPointContainer 
                mode="normal"
                icons={icons}
                titles={titles}
                texts={texts}
                color="yellow"/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('renders BulletPointContainer in expanded mode', () => {
        const wrapper = shallow(
            <BulletPointContainer 
                mode="expanded"
                icons={icons}
                titles={titles}
                texts={texts}
                color="yellow"/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});