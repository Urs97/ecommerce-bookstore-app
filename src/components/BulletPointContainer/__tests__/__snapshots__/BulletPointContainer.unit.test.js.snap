// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletPointContainer renders BulletPointContainer in expanded mode 1`] = `
<section
  className="bulletpoints expanded"
>
  <BulletPoint
    color="yellow"
    icon="mockIcon1.svg"
    mode="expanded"
    text="This is text number 1"
    title="Title 1, Title 2, Title 3"
  />
  <BulletPoint
    color="yellow"
    icon="mockIcon2.svg"
    mode="expanded"
    text="This is text number 2"
  />
  <BulletPoint
    color="yellow"
    icon="mockIcon3.svg"
    mode="expanded"
    text="This is text number 3"
  />
</section>
`;

exports[`BulletPointContainer renders BulletPointContainer in normal mode 1`] = `
<section
  className="bulletpoints"
>
  <BulletPoint
    color="yellow"
    icon="mockIcon1.svg"
    mode="normal"
    text="This is text number 1"
  />
  <BulletPoint
    color="yellow"
    icon="mockIcon2.svg"
    mode="normal"
    text="This is text number 2"
  />
  <BulletPoint
    color="yellow"
    icon="mockIcon3.svg"
    mode="normal"
    text="This is text number 3"
  />
</section>
`;

exports[`BulletPointContainer renders BulletPointContainer in unrecognizable mode 1`] = `
<div>
  BulletPoint error
</div>
`;
